### ISSUE TITLE
Singleton scope in Play module is not mapped as Scaldi singleton scope?

### ISSUE THREAD

**Issue by andrey-chernyshev** [scaldi/scaldi#62]
I'm using ReactiveMongo Play module which is defined as follows:

    @Singleton
    final class ReactiveMongoModule extends Module {
      override def bindings(environment: Environment, configuration: Configuration): Seq[Binding[_]] =    Seq(bind[ReactiveMongoApi].to[DefaultReactiveMongoApi].in[Singleton])
    }

However, each time I'm injecting [ReactiveMongoApi]  I get a new instance:

    import com.typesafe.config.{ConfigFactory, ConfigResolveOptions}
    import play.api.Environment
    import play.modules.reactivemongo.ReactiveMongoApi
    import scaldi.Injectable
    import scaldi.play.ScaldiApplicationBuilder
    import scala.collection.JavaConversions._

    object Test extends App with Injectable {
  
      val conf = play.api.Configuration.from(ConfigFactory.load("test.conf")     .resolve(ConfigResolveOptions.defaults()).entrySet().map(e => (e.getKey, e.getValue.unwrapped)).toMap)

     implicit val injector = new ScaldiApplicationBuilder(Environment.simple(), conf).buildInj()
     val rm1 = inject[ReactiveMongoApi]
     val rm2 = inject[ReactiveMongoApi]
     println(rm1 == rm2)
    }

- the test will be printing "false".  Shouldn't the rm1 & rm2 instances be same?

To run the test, include ReactiveMongo with the build:
 
     "org.reactivemongo" %% "play2-reactivemongo" % "0.11.7.play24"   
    
and put  a "test.conf" file under the /conf directory with the following lines:

    play {
      modules = {
        enabled += "play.modules.reactivemongo.ReactiveMongoModule"
      }
   }

**Comment by Mironor** [comment]
Hi Andrey,

why are you using Guice's module definition instead of Scaldi's one?

And I think it's because you are binding ReactiveMongoApi to a provider function instead of an instance. Provider executes the function (so creates a new instance) on every injection.

Try this solution:

```
class ReactiveMongoModule extends Module {
  bind[ReactiveMongoApi] to new DefaultReactiveMongoApi(inject[ActorSystem], inject[Configuration], inject[ApplicationLifecycle])
}
```

**Comment by Mironor** [comment]
Don't forget to enable play's modules in applicaiton.conf:

```
play.application.loader = scaldi.play.ScaldiApplicationLoader
play.modules.enabled += "scaldi.play.ControllerInjector"
```

and in test.conf
```
play.modules.enabled += "play.modules.reactivemongo.ReactiveMongoModule"
```

**Comment by andrey-chernyshev** [comment]
Hi Alexandre, 
I forgot to mention that the ReactiveMongo module dfinition I wrote comes with the ReactiveMongo package.  I can workaround this by manually creating Scaldi module as you suggest, but it would be nice if I can take and use the Play module as is, as long as Scaldi supports combining with Play modules.

**Comment by Mironor** [comment]
So your ReactiveMongoApi is already injected by ReactiveMongo plugin, you are trying to override the injection in your tests, right?

```
class TestModule extends Module {
  bind[ReactiveMongoApi] to new DefaultReactiveMongoApi(inject[ActorSystem], inject[Configuration], inject[ApplicationLifecycle])
}
```

for test.conf doesn't cut it for you ? (I changed module's name)

**Comment by andrey-chernyshev** [comment]
No, I'm not overring anything.  All code I wrote is in the Test.  I'm just injecting the ReactiveMongoApi that was originally bound in the ReactiveMongo Play module.

**Comment by Mironor** [comment]
You declared injection as a provider, not as binding, so every time you use `inject` it creates a new instance of DefaultReactiveMongoApi: http://scaldi.org/learn/#provider-binding

**Comment by Mironor** [comment]
Could you provide the play example in a github (or any other) repository so that I could try to see the problem myself?

**Comment by andrey-chernyshev** [comment]
I have assembled the minimized test at https://github.com/andrey-chernyshev/Scaldi62.
You can try running a Test example from there, please let me know if I'm doing anything wrong.
Thanks for looking into this!

**Comment by OlegIlyenko** [comment]
Thanks for creating an example application that reproduces this problem! On the first sight it looks like a bug since `ResctiveMongoApi` is bound as a `providerBinding` even though it's bound as a singleton in the play module. 

@Mironor thanks a lot for looking into it as well. I agree, a temporary workaround would be to bind `ResctiveMongoApi` manually. I will look at the issue later today

**Comment by OlegIlyenko** [comment]
Indeed, the singleton scope was not considered for some types of play bindings. I just released scaldi-play v0.5.11 with the bugfix. It should be shortly available for download from the maven central.

I would appreciate if you could confirm that new version fixed the issue in your application.
